### PR TITLE
Add signature, signer, price hash, hash version, quote liquidity

### PR DESCRIPTION
**Pull request by gagiuntoli**
## Context

Add signature, signer, price hash, hash version, quote liquidity to Price Data Model

Document example:

```json
{
  "_id": {
    "$oid": "66a0d4928893836b157bb38e"
  },
  "timestamp": 1721816210,
  "feedBase": "WRBTC",
  "feedQuote": "DOC",
  "fetcherSource": "",
  "valueType": "Price",
  "fetcher": "CoingeckoPrice",
  "value": "1.007",
  "hashVersion": 1,
  "signature": "0x0aec957b9f3ba5b46b1f901af8f7a03ee2564f6cd229b6f126b3e689e0c077a70f8d3c0864e1c6b3fd354e809ce48183ae3088785dce53e6f8eebae3403726dc1c",
  "signer": "0x7E5F4552091A69125d5DfCb7b8C2659029395Bdf"
}
```

## To-do list

- [ ] Added tests
- [ ] Tested on dev/sbx
- [ ] Changelog was updated with current changes
- [ ] Documentation or guides were updated (slab, readme)
- [ ] Version was updated on `package.json` (releases/hotfixes)

## Checklist before merge

- [ ] **there are no errors in logs in any workers**
- [ ] new blocks are being discovered and marked as finalized
- [ ] foreign blocks are being dispatched
- [ ] squash all commits before merging

### CODE REVIEW

File: src/models/PriceDataModel.ts [hunk 1]
```diff
@@ -27,4 +28,19 @@ export class PriceDataModel {
 
   @prop({required: false})
   fetcherSource!: string;
+
+  @prop({required: true})
```
Comment (DZariusz) [inline src/models/PriceDataModel.ts:32]: it is not required, it is optional only for dexes

**Review by DZariusz — CHANGES_REQUESTED** [review]
this PR has only model fields, what about functionality?

File: src/services/PriceSignerService.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import {ChainsIds} from '../types/ChainsIds.js';
+
+@injectable()
+class PriceSignerService {
+  @inject(DeviationSignerRepository) protected deviationSignerRepository!: DeviationSignerRepository;
+
+  async sign(message: string): Promise<{signerAddress: string; hashVersion: number; signature: string; hash: string}> {
+    const hashVersion = 1;
+    const hash = ethers.utils.id(message);
+    const signer = this.deviationSignerRepository.get(ChainsIds.ROOTSTOCK);
```
Comment (DZariusz) [inline src/services/PriceSignerService.ts:14]:
```suggestion
    const signer = this.deviationSignerRepository.get(ChainsIds.ETH);
```
lets use ETH, this is our main blockchain.

**Review by DZariusz — APPROVED** [review]

**Review by DZariusz — CHANGES_REQUESTED** [review]
test is missing, please check jira task for details

File: src/models/PriceDataModel.ts (hunk 1 above)
Comment (gagiuntoli) [reply src/models/PriceDataModel.ts]: done

File: src/repositories/PriceDataRepository.ts [hunk 3]
```diff
@@ -36,13 +41,24 @@ export class PriceDataRepository {
...
-      },
-    }));
+    const deviationSigner = this.deviationSignerRepository.get(ChainsIds.ROOTSTOCK);
+    const signer = await deviationSigner.address();
+    const hashVersion = 1;
+
+    const bulkOps = await Promise.all(
+      data.map(async (doc) => {
+        const newDoc = {...doc, hashVersion};
+        const priceHash = ethers.utils.id(String(newDoc));
```
[the remark this answers is not in the document]
Comment (gagiuntoli) [reply src/repositories/PriceDataRepository.ts]: done

File: src/repositories/PriceDataRepository.ts [hunk 4]
```diff
@@ -36,13 +41,24 @@ export class PriceDataRepository {
...
     const model = await getModelForClass(PriceDataModel);
 
-    const bulkOps = data.map((doc) => ({
-      updateOne: {
-        filter: {...doc},
-        update: doc,
-        upsert: true,
-      },
-    }));
+    const deviationSigner = this.deviationSignerRepository.get(ChainsIds.ROOTSTOCK);
```
[the remark this answers is not in the document]
Comment (gagiuntoli) [reply src/repositories/PriceDataRepository.ts]: done

**Review by DZariusz — APPROVED** [review]

**Pull request merged**